### ISSUE TITLE
[BUG] Unexpected behavior of --run-dir

### ISSUE THREAD

**Issue by konnov** [informalsystems/apalache#1122]
## Description

Run apalache in its home directory with the `--run-dir` specified:

```
apalache-mc check --run-dir=foo --inv=BuggyStateInv ./test/tla/Invariants.tla
```

This creates two directories:

 - `foo` (as expected)
 - `_apalache-out/Invariants.tla/2021-11-25T10-52-02_2923373664224331046` (unexpected)

Moreover, the model checker points to counterexamples in `_apalache-out`, not in `foo`:

```
State 1: state invariant 0 violated. Check the counterexample in: 
  /Users/igor/devl/informal/audits-internal/ethermint2110/specs/_apalache-out/Invariants.tla/2021-11-25T10-52-02_2923373664224331046/counterexample1.tla
```

I would expect apalache to create only one directory instead of two, and this should be the directory as specified with `--run-dir`. Otherwise, we are back to cluttering the file system.

## Input specification

`./test/tla/Invariants.tla` in the apalache home directory.

**Comment by shonfeder** [comment]
This is the intended behavior. But maybe we need to have a discussion about what the intended use of the default outside is to get clear on what we’re doing.

**Comment by shonfeder** [comment]
Here's some context:

the `--help` documents this argument as follows:

```
   --run-dir            : additional directory wherein output files for this run will be written directly, default: none (overrides envvar RUN_DIR)
```

The *additional* is meant to communicate that this an additional, stable, location where the output files will be written. 

The issue that posed the need for being able to write output to a stable and known location was #1077 and the initial design proposed there (https://github.com/informalsystems/apalache/issues/1077#issuecomment-961279449) explicitly proposes added an additional, stable location to which the outputs would be written. That included a proposal for writing just the principle output files from a run to a known file, but this won't work for counter examples because we write `n` different files (I don't actually know why we do this). So, instead, the proposed `output` CLI argumet was replaced with the new `run-dir` flag, which **adds** a stable, known, location, but does not change the usual behavior of writing to the `_apalache_out`.

My current assumptions:

- I've been assuming that we need `_apalache_out` to gather all run outputs for some sort of historical purposes. If not, then I'm unclear on why we are writing all of this output (the logs and run files etc.) when users don't ask for it.
- I assumed that we do not need the location of the counterexample in the `run-dir` included in the message `Check the counterexample in` because users will already know exactly where the `run-dir` is located, having the explicitly set it themselves.

**Comment by konnov** [comment]
Right. As we discussed, we should probably extend the ADR with the user scenarios and check whether it makes sense for the user (and scripts) to output files like that

**Comment by shonfeder** [comment]
Agreed! 

I think it's pretty clear that we don't currently have a coherent and systematic understanding of what our various output files are used for, where they should be written, or when they should be generated.

As per our discussions earlier, I think the next action items are: 

- [ ] Revisit the output ADR, particularly with an eye to giving a systematic account of how each of the outputs in the different categories is *used*
- [ ] Propose a revamp of the output schema that is anchored in
   - The use cases identified through the previous analysis
   - Predominating unix conventions
   - Any post-unix critiques that advise improved ways of organizing and exposing outputs
- [ ] Once we've reached agreement on the revamp, implement it

I would argue that we *don't* want to simply have `--run-dir` override `--out-dir`, as 

1. this creates clashing commands in the CLI and, 
2. one of the overlapping uses currently made of the default outdir is to collect a historical archive of all the logs, and introducing behavior that causes these logs to sometimes be written and sometimes not would be quite strange.
